mediator-coordination/src/web/mod.rs

```diff
@@ -1,7 +1,94 @@
-use axum::Router;
+mod coord;
+mod error;
+mod handler;
+mod midlw;
+
+use axum::{middleware, routing::post, Router};
 use did_endpoint::util::keystore::KeyStore;
-use did_utils::didcore::Document;
+use did_utils::{didcore::Document, key_jwk::jwk::Jwk};
+use std::sync::Arc;
+
+use crate::{
+    didcomm::bridge::{LocalDIDResolver, LocalSecretsResolver},
+    model::stateful::coord::entity::Connection,
+    model::stateful::coord::entity::Secrets,
+    repository::traits::Repository,
+    util,
+};
 
-pub(crate) fn routes(_diddoc: Document, _keystore: KeyStore) -> Router {
+pub fn routes(state: Arc<AppState>) -> Router {
     Router::new()
+        // Unified route for all DIDComm messages
+        .route("/", post(handler::process_didcomm_message))
+        .route_layer(middleware::from_fn_with_state(
+            state.clone(),
+            midlw::unpack_didcomm_message,
+        ))
+        // Transitive routes
+        .route(
+            "/mediate",
+            post(coord::handler::process_didcomm_mediation_request_message),
+        )
```

Comment:
Drop this route, but ensure that the stateful mediation request is channeled through the unified route.

Reply:
Ok sir